blockchain/vm/internaltx_tracer.go

```diff
@@ -515,24 +515,57 @@ func (this *InternalTxTracer) result() (*InternalTxTrace, error) {
 	}
 	if err := this.ctx["error"]; err != nil && err.(error).Error() == errEvmExecutionReverted.Error() {
 		outputHex := this.ctx["output"].(string) // it is already a hex string
-		if len(outputHex) >= 11 && outputHex[2:10] == "08c379a0" {
+		outputHexLength := len(outputHex)
+
+		// outputHexLength should be equal for larger than 138 (10+32*2+32*2) to parse a revert string
+		// outputHex[:10]: "0x08c379a0"
+		// outputHex[10:10+32*2]: a revert string
+		// outputHex[10+32*2:10+32*2+32*2]: the length of a revert string
+		if outputHexLength >= 138 && outputHex[2:10] == "08c379a0" {
 			defaultOffset := 10
 
-			stringOffset, err := strconv.ParseInt(outputHex[defaultOffset:defaultOffset+32*2], 16, 64)
+			// TODO-Klaytn: introduce error handling logic for the case the parsing data is bigger than math.MaxUint64
+			stringOffset, err := strconv.ParseUint(outputHex[defaultOffset:defaultOffset+32*2], 16, 64)
 			if err != nil {
 				logger.Error("failed to parse hex string to get stringOffset",
 					"err", err, "outputHex", outputHex)
 				return nil, err
 			}
-			stringLength, err := strconv.ParseInt(outputHex[defaultOffset+32*2:defaultOffset+32*2+32*2], 16, 64)
+			stringLength, err := strconv.ParseUint(outputHex[defaultOffset+32*2:defaultOffset+32*2+32*2], 16, 64)
 			if err != nil {
 				logger.Error("failed to parse hex string to get stringLength",
 					"err", err, "outputHex", outputHex)
 				return nil, err
 			}
+
+			if stringOffset > uint64(outputHexLength) {
+				stringOffset = 0
+			}
+			if stringLength > uint64(outputHexLength) {
+				stringLength = 0
+			}
+
 			start := defaultOffset + 32*2 + int(stringOffset*2)
 			end := start + int(stringLength*2)
-			asciiInBytes, err := hex.DecodeString(outputHex[start:end])
+
+			if start < 0 {
+				start = 0
+			}
+			if end < 0 {
+				end = 0
+			}
+			// return nothing if end is out of the range
```

Comment:
@aidan-kwon What if start > outputHexLength ?

Reply:
`end := start + int(stringLength*2)` and stringLength is always positive or zero. So, end > outputHexLength if start > outputHexLength

Reply:
int(stringLength*2) can be negative.

Reply:
stringLength is equal or smaller than outputHexLength. outputHexLength is a slice size, and the size cannot be bigger than `MaxInt/2`

Reply:
okay. got it. Thanks.